### PR TITLE
feat: Prober for black-box monitoring of archival nodes

### PR DESCRIPTION
**Pull request by nikurt**

### CODE REVIEW

File: pytest/tools/prober/prober.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        # missing because the node doesn't have history, or because the block
+        # was skipped.
+        if not block:
+            if random_height == genesis_height:
+                logger.error(f'Genesis block is missing. This is impossible {random_height}')
+
+            random_height -= 1
+            attempt += 1
+            # Limit the number of attempts.
+            if attempt > 10:
```
Comment (posvyatokum) [inline pytest/tools/prober/prober.py]: I think at least during the October testnet incident we missed a bunch of blocks in a row. Maybe should add some exceptions here later.

**Review by posvyatokum — APPROVED** [review]
Looks good, can start this and in the meantime check testnet and mainnet db for large holes in heights using ProcessedBlockHeights or something

File: pytest/tools/prober/prober.py (hunk 1 above)
Comment (nikurt) [reply pytest/tools/prober/prober.py]: Statistically that is not going to be an issue. Configuring alerting to 90% will allow for the October testnet incident.

File: pytest/tools/prober/prober.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        j = {
+            'method': method,
+            'params': params,
+            'id': 'dontcare',
+            'jsonrpc': '2.0'
+        }
+        start_time = time.time()
+        r = requests.post(url, json=j, timeout=5)
+        latency_ms = (time.time() - start_time)
+        print(f'prober_request_latency_ms{{method="{method}",url="{url}"}} {latency_ms:.2f}')
```
Comment (mm-near) [inline pytest/tools/prober/prober.py:40]: is this for cloudprober?

File: pytest/tools/prober/prober.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+"""
+Prober that is compatible with cloudprober.
+Prober verifies that any block since the genesis can be retrieved from a node.
```
Comment (mm-near) [inline pytest/tools/prober/prober.py:5]: s/any/random

**Review by mm-near — APPROVED** [review]

File: pytest/tools/prober/prober.py (hunk 2 above)
Comment (nikurt) [reply pytest/tools/prober/prober.py]: This metric is re-exported by cloudprober and it's available in Grafana.

**Pull request merged**